pyhap/service.py

```diff
@@ -68,3 +71,15 @@ def to_HAP(self):
             'type': str(self.type_id).upper(),
             'characteristics': [c.to_HAP() for c in self.characteristics],
         }
+
+    @classmethod
+    def from_dict(cls, name, json_dict, char_loader):
+        """Initialize a service object from a dict."""
+        type_id = UUID(json_dict.pop('UUID'))
+        service = cls(type_id, name)
+        chars = []
+
+        for name in json_dict['RequiredCharacteristics']:
+            chars.append(char_loader.get(name))
```

Comment:
Is there a reason to prefer this to
```python
for name in json_dict[...]:
    service.add_characteristic(char_loader.get(name))
```
Personally I think this is more readable

Reply:
I agree. Must have been so obvious that I've overlooked it.